packages/utils/src/ts-eslint/Config.ts

```diff
@@ -0,0 +1,244 @@
+  export type RuleLevelAndOptions = SharedConfig.RuleLevelAndOptions;
+  export type RulesRecord = SharedConfig.RulesRecord;
+  export type Severity = SharedConfig.Severity;
+  export type SeverityString = SharedConfig.SeverityString;
+
+  // https://github.com/eslint/eslintrc/blob/v2.1.0/conf/config-schema.js
+  interface BaseConfig {
+    $schema?: string;
+    /**
+     * The environment settings.
+     */
+    env?: EnvironmentConfig;
+    /**
+     * The path to other config files or the package name of shareable configs.
+     */
+    extends?: string[] | string;
+    /**
+     * The global variable settings.
+     */
+    globals?: GlobalsConfig;
+    /**
+     * The flag that disables directive comments.
+     */
+    noInlineConfig?: boolean;
+    /**
+     * The override settings per kind of files.
+     */
+    overrides?: ConfigOverride[];
+    /**
+     * The path to a parser or the package name of a parser.
+     */
+    parser?: string | null;
+    /**
+     * The parser options.
+     */
+    parserOptions?: ParserOptions;
+    /**
+     * The plugin specifiers.
+     */
+    plugins?: string[];
+    /**
+     * The processor specifier.
+     */
+    processor?: string;
+    /**
+     * The flag to report unused `eslint-disable` comments.
+     */
+    reportUnusedDisableDirectives?: boolean;
+    /**
+     * The rule settings.
+     */
+    rules?: RulesRecord;
+    /**
+     * The shared settings.
+     */
+    settings?: SharedConfigurationSettings;
+  }
+
+  export interface ConfigOverride extends BaseConfig {
+    excludedFiles?: string[] | string;
+    files: string[] | string;
+  }
+
+  export interface Config extends BaseConfig {
+    /**
+     * The glob patterns that ignore to lint.
+     */
+    ignorePatterns?: string[] | string;
+    /**
+     * The root flag.
+     */
+    root?: boolean;
+  }
+}
+
+export namespace FlatConfig {
+  export type EcmaVersion = ParserOptionsTypes.EcmaVersion;
+  export type GlobalsConfig = SharedConfig.GlobalsConfig;
+  export type Parser = ParserType.ParserModule;
+  export type ParserOptions = SharedConfig.ParserOptions;
+  export type Processor = ProcessorType.ProcessorModule;
+  export type RuleEntry = SharedConfig.RuleEntry;
+  export type RuleLevel = SharedConfig.RuleLevel;
+  export type RuleLevelAndOptions = SharedConfig.RuleLevelAndOptions;
+  export type Rules = SharedConfig.RulesRecord;
+  export type Settings = SharedConfigurationSettings;
+  export type Severity = SharedConfig.Severity;
+  export type SeverityString = SharedConfig.SeverityString;
+  export type SourceType = ParserOptionsTypes.SourceType | 'commonjs';
+
+  export interface Plugin {
+    /**
+     * The definition of plugin processors.
+     * Can be referenced by string in the config (i.e., `"pluginName/processorName"`).
+     */
+    processors?: Record<string, Processor>;
+    /**
+     * The definition of plugin rules.
+     */
+    rules?: Record<string, RuleCreateFunction | AnyRuleModule>;
+  }
```

Comment:
https://eslint.org/docs/latest/extend/plugins#meta-data-in-plugins

```suggestion
  export interface Plugin {
    /**
     * The definition of plugin processors.
     * Can be referenced by string in the config (i.e., `"pluginName/processorName"`).
     */
    processors?: Record<string, Processor>;
    /**
     * The definition of plugin rules.
     */
    rules?: Record<string, RuleCreateFunction | AnyRuleModule>;
    /**
     * Metadata about your plugin for easier debugging and more effective caching of plugins
     */
    meta?: {
      /**
       * The meta.name property should match the npm package name for your plugin.
       */
      name: string,
      /**
       * The meta.version property should match the npm package version for your plugin.
       */
      version: string,
    },
  }
```

Reply:
There's also https://eslint.org/docs/latest/extend/plugins#configs-in-plugins that wasn't added.
I was waiting on them to update the docs to show what the new configs look like but IDK when they'll do that.........

Reply:
Probably need to check these types against https://eslint.org/docs/latest/extend/plugin-migration-flat-config to double check they are correct.

I originally wrote them blindly slash in conjunction with `@types/eslint` and the eslint source code.